controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -258,7 +258,19 @@ func (r *KubeadmControlPlaneReconciler) reconcile(ctx context.Context, cluster *
 		return r.upgradeControlPlane(ctx, cluster, kcp, ownedMachines, requireUpgrade)
 	}
 
-	// If we've made it this far, we we can assume that all ownedMachines are up to date
+	// If we've made it this far, we can assume that all control-plane machines are up to date
+	workloadCluster, err := r.managementCluster.GetWorkloadCluster(ctx, util.ObjectKey(cluster))
+	if err != nil {
+		logger.Error(err, "failed to get remote client for workload cluster", "cluster key", util.ObjectKey(cluster))
```

Comment:
```suggestion
		logger.Error(err, "failed to get remote client for workload cluster")
```

the logger should have enough info already